veps/sig-network/hotpluggable-nad-ref.md

```diff
@@ -0,0 +1,195 @@
+# VEP #140: Live Update NetworkAttachmentDefinition Reference
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [ ] (R) Target version is explicitly mentioned and approved
+- [ ] (R) Graduation criteria filled
+
+## Overview
+
+VMs using secondary networks refer to a NetworkAttachmentDefinition object's namespace and/or name as part of their network specification. 
+Currently, the whole network specification, including this reference - is immutable for the lifecycle of a VM.
+If a user wishes to change the network a VM is connected to (example: switching VLANs), they must modify the VM specification and restart the VM. 
+This proposal introduces the ability to update the NAD reference (networkName) of a network on a running VM through Live Migration. 
+This allows the host network connection to change without any reboot.
+
+## Motivation
+
+Currently, in order to move VMs from one network to another, the user either needs to hotplug a new interface and unplug the old one or updated the NAD reference and reboot the VM.
+We want to implement a way to do this without reboot and without a change in the guest interfaces.
+
+## Goals
+
+- Change an existing NAD reference (`spec.networks[].multus.networkName`) on a running VM.
+
+## Non Goals
+
+- Migrating between incompatible CNI types
+- Verifying that the NAD corresponding to the new NAD reference exists
```

Comment:
Actually, this is currently done implicitly every time a virt-launcher pod is templated [1] (for a new VM or for a migration target).

Please consider dropping this non-goal.

[1] https://github.com/kubevirt/kubevirt/blob/4ee5ee2813a7170686e7d6f855fbc5f334de36d6/pkg/virt-controller/services/template.go#L366